src/core/qgsrendercontext.cpp

```diff
@@ -359,3 +405,8 @@ double QgsRenderContext::convertFromMapUnits( double sizeInMapUnits, QgsUnitType
   }
   return 0.0;
 }
+
+double QgsRenderContext::convertMetersToMapUnits( double meters ) const
+{
+  return mDistanceArea.measureLineProjected( mExtent.center(), meters );
```

Comment:
I'd suggest adding a shortcut here if the destination CRS units are already meters